invenio/modules/workflows/client.py

```diff
@@ -39,15 +44,25 @@ def run_workflow(wfe, data, stop_on_halt=False,
     :param wfe: workflow engine in charge of workflow execution
     """
     while True:
+        # Make a backup of the db session and replace the global one with a
+        # new clean version. This is required because the workflow payload
+        # might do horrible things with it and we need to for querying+storing
+        # the workflow state/objects.
+        session_backup = db.session
+        db.session = db.create_scoped_session()
```

Comment:
Does this physically creates a new connection to the DB? Or are connections taken from a fixed-size pool? Otherwise we risk to exhaust the available DB connections (e.g. in case of many parallel workflows).